### DISCUSSION TITLE
Shelly 1PM with ESP-XXXXXX SSID

### DISCUSSION THREAD

**Discussion by host-felice** [arendst/Tasmota#17523] [Q&A]
So, I have some shelly 1PM flashed tasmota (OTA Mongoose to Tasmota) and they work fine.
All of a sudden, one of them went offline. I tried to restart it: no luck. 7 power cycles to reset it: instead of tasmota-xxxxxx I got ESP-xxxxxx. The webpage at 192.168.4.1 goes in timeout.

Flashing from scratch via serial adapter works, but ONLY keeping connected the device to the 3,3V serial adapter. With 220V it doesn't work. I tought of a problem with the AC-DC converter, but...

Out of curiosity, I restored the original shelly firmware and it works perfectly both 3,3V and 220V. Then again changed to tasmota with mongoose to tasmota ota and again it doesn't work.

I am very confused, I tried different tasmota firmwares, but still it doesn't work on 220V with tasmota, no problem with the original software. I don't know how to investigate since I cannot connect the serial adapter while powering from the 220V.

Any idea?
Thanks

**Comment by sfromis** [comment]
`ESP-xxxxxx` means bricked, Tasmota not available, you need to flash again from the start. Don't know what happened.....

**Comment by host-felice** [comment]
well, ok... BUT:
1) if I connect it to the 3,3V it shows up as tasmota-xxxxxx
2) then again to the 220V and it shows up as ESD-xxxxxx

so, it is not bricked

If I flash from scratch with tasmota and them back to the original shelly firmware, it works both at 3,3V and 220V. If I return to tasmota @220V it doesn't work again and the reset doing 7 power cycles brings to the ESD-xxxxxx ssid again.

**Reply by sfromis** [reply]
Don't know what happens. I've had no such trouble with my Shelly energy monitoring devices (not specifically 1PM, but other models like 2.5 and 1PM+, 2PM+). Doing anything unusual with wiring? Does it also happen if you try things out without a load/switch connected?

**Comment by host-felice** [comment]
I did nothing special, it was connected like other 1PM, 2.5 and EM I have in the house, all of them with tasmota for more than 1 year.

With tasmota, it works only @3,3V. I thought it was the AC-DC converter, but it works with the original shelly firmware both 3,3V and 220V.

I really don't understand how it is possible. I reverted back to 12.2.0 just in case, but no luck.

**Comment by host-felice** [comment]
Updated the original post: with tasmota it works only at 3,3V from the serial adapter. I tried different tasmota versions, no luck. With the original shelly firmware (flashed from scratch with serial adapter), it works at 220V perfectly.

It is a little bit crazy and I really don't know how to investigate.

The only thing I noticed, with tasmota firmware, my shellys do a very fast turn on/turn off of the internal relay while booting (and I don't like it), with the shelly firmware they don't. Maybe there is something involved with it. The internal relay is wired to GPIO15.

**Comment by arendst** [comment]
This might well be a power supply problem shellies ar known for in relation to Tasmota features.

You might try one (or more) the following:
- delay initialization using ``SO46 1..255`` for 10..2550 milliseconds
- delay tasmota relay control at power on using ``SO47 1`` - Wait for wifi, ``SO47 2`` - Wait for MQTT or ``SO 3..255`` - Wait 3..255 seconds
- use tasmota-lite.bin

**Comment by host-felice** [comment]
I tried the latest dev to solve the gpio15 problem. Very high delay, wait for mqtt, tasmota-lite... nothing. I still get the ESP-XXXXXX when @220V.

Is there something else I can try? Or do I have to gave up on this 1PM and tasmota and use it with the original firmware only?

**Reply by arendst** [reply]
As your other 1PM works just fine we ran out of Tasmota solutions.

**Comment by host-felice** [comment]
yep, I have 3 1PMs with tasmota and they work just fine. The 4th worked like a charm too, not sure what happened :-( I really don't like the shelly firmware, expecially the mqtt...

Anyway, thanks for your support. I am just curious to understand what is inside the shelly original firmware to make it possible to work where tasmota can't. Maybe something to use less energy? Or delay the boot somehow. The original firmware blinks the led for some seconds and then stays on when connected to wifi. Tasmota seems to be faster connecting to wifi and it doesn't blink the led, so it should be a little bit less power hungry without the led.

ODD.

**Reply by sfromis** [reply]
SetOption46 and SetOption47 were created to mitigate the potential power spike when booting Tasmota. Did you try playing with some values here?

**Comment by host-felice** [comment]
yep. I tried 100, 150 and 200 in different combination. I still get the ESP-XXXXXX AP

**Comment by Syntaxrabbit** [comment]
I got two brand new Shelly 1PM today, same issue: Tasmota PyFlasher works, but the result is always the AP named "ESP-XXXXXX", no matter which (old) version and variant of Tasmota I try to flash. Ideas to investigate?

**Reply by sfromis** [reply]
In that case, your conclusion that "Tasmota PyFlasher works" does not have much of a basis, when you get the `ESP-xxxxxx` name. That is generally what you get if the chip was not able to boot into the firmware, for some unclear reason. I can't see what went wrong, but I'd suggest instead trying the recommended Tasmota web installer, allowing it to detect the chip and automatically select matching firmware. And after flashing, you can run a serial terminal program to follow if Tasmota is able to boot, and what happens.

**Comment by host-felice** [comment]
The problem is we cannot check the serial boot when connected at 220V

**Comment by Syntaxrabbit** [comment]
OK, after testing lots of flashers and Tasmota versions, I got to it to at least create the Wifi AP, but as soon as I try to connect to my IOT network, the device crashes.

(Tasmota Version 12.3.1(lite)-2_7_4_9)

[crashlog_from_console.log](https://github.com/arendst/Tasmota/files/10470438/crashlog_from_console.log)

**Comment by host-felice** [comment]
I resume this old post cause I was able to flash esphome on my 1pm. It works @220V. So, after using it for few days, I tried to go back to tasmota. I uploaded the lite version and tasmota was up and running @220V. Moved to tasmota regular version and again it worked like before. I used it for few days and doing some upgrade meanwhile. So far so good. Today I wanted to try a self compiled version with tasmocompiler: minimal -> my build. It worked. Then I realized I forgot to add one thing: minimal... and it didn't go back online. It was rebooting in a loop, I heard the relay clicking after a while.

So, I spent the whole afternoon trying to get it back to tasmota, no success. Now it works at least with the lite version, which is an improvement of the previous situation.

The odd thing is it did work with the regular tasmota for days @220V. No problem at all.

Now I upgraded from lite to tasmota.bin and it worked @220V but the web server is slow. I send an ON command and the relay turns on. Off command and the device slows down and starts to reboot in loop.

I am very sad... I think I will have to use it with esphome :-(

**Comment by host-felice** [comment]
Info3":{"RestartReason":{"Exception":28,"Reason":"Blocked Loop","EPC":["402054ee","00000000","00000000"],"EXCVADDR":"00000000","DEPC":"00000000","CallChain":["402054ec","4010605d","4024ed3e","40275c78","4024fd02","40275c85","40275cca","402495a4","40000f49","40000f49","40000e19","401059fd","40105a03","4010000d","4026e6fc","4026e6ad","40101649","40101600","4000050c","4000da01","4000da01","4028ae67","40101649","40101600","4000050c","40253815","4025381a","40253c18","40253c18","40253ce2","40254619"]},"BootCount":21}}

**Comment by host-felice** [comment]
ok, something odd: SleepMode":"Dynamic","Sleep":50,"LoadAvg":1098

the loadavg is way too high.

**Comment by Bodengriller** [comment]
Same thing[ here ](https://github.com/arendst/Tasmota/discussions/20210)
I have 2 Shelly 1PM - one wirks with Tasmota 10 and highter without any problems.
The other one works with tasmota up to V10.0 without any problems, too.
But when I flash V11.1 or higher, this Shelly 1PM only shows up WLAN "ESP-060496" and is not working anymore.
If I connect this Shelly directly to 3.3V instead of 230V, it is working even with Tasmota V13.2.

The other Shelly 1PM does his job with V13.2 and 230V without any problems.

I don't know what else to do but switch to ESPhome to keep all Shellys on the same Frimware level. :(
Or downgrade all Shellys to V10.0 - which I don't like at all.
